### PR TITLE
#3128561 - Hook cron api

### PR DESCRIPTION
**Pull request by shabananavas**

### CODE REVIEW

File: entity_sync.module [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    return $items;
+  }
+
+  // Go through the entity types that need to be synced and create a cron item
+  // for each entity type.
+  foreach ($entities_to_sync as $sync_id => $entity_info) {
+    $item_name = $entity_info['provider']
+      . '_' . $entity_info['entity_type_id']
+      . '_' . $entity_info['entity_bundle'];
+    $items[$item_name] = _entity_sync_create_cron_job($sync_id, $entity_info);
```
Comment (krystalcode) [inline entity_sync.module]: @shabananavas The `sync_id` is the unique identifier of the sync definition, right? For example `entity_sync.sync.user`. That should then be unique so you can use that together with the operation the job will do i.e. `$items[$sync_id . '_import_list'] = ...`.

**Review by krystalcode — CHANGES_REQUESTED** [review]

File: entity_sync.module (hunk 1 above)
Comment (shabananavas) [reply entity_sync.module]: Yup.

**Comment by shabananavas** [comment]
@krystalcode I was busy writing away the code and didn't bother to test if hook_cronapi actually works. In D8, I don't think we use hook_cronapi to declare the cron jobs: https://www.drupal.org/project/ultimate_cron/issues/2744129

**Comment by krystalcode** [comment]
@shabananavas 

Looked into it, the situation seems pretty complicated; there are a couple of solutions but need a bit of an investigation. We'll do that when we have the time to do so.

For now, we'll have to define the jobs as Ultimate Cron configuration files at the application level. For example: `ultimate_cron.job.entity_sync_${SyncType}`, where `${SyncType}` the ID of the sync type.

Unfortunately, passing arguments to the job is not supported; so (for now) we'll have to define a callback at the application level e.g. `MY_MODULE_entity_sync_${SyncType}` which will call a service provided by `entity_sync` with the right arguments.

I'm closing this MR as things change drastically.

**Pull request closed without merging**